### PR TITLE
database: add class

### PR DESCRIPTION
**Pull request by MrColorado**

### CODE REVIEW

File: src/entity/Store.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  position: object;
+
+  @ManyToOne(type => Brand, brand => brand.id)
+  brand: Brand;
+
+  @OneToMany(type => StorePromotion, storePromotion => storePromotion.id)
+  storePromotion: StorePromotion[];
+
+  @ManyToMany(type => User, user => user.id)
+  manager: User[];
```
[the remark this answers is not in the document]
Comment (MrColorado) [reply src/entity/Store.ts]: What do you mean ? A user can be a manager of many stores and there can be many managers for a store

File: src/entity/UsedList.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  Entity,
+  PrimaryColumn,
+  Column,
+  PrimaryGeneratedColumn,
+  OneToOne,
+} from 'typeorm';
+import { List } from './List';
+
+@Entity()
+export class UsedList {
```
[the remark this answers is not in the document]
Comment (MrColorado) [reply src/entity/UsedList.ts]: List that have already be used and when it has been used

File: src/entity/Store.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugodelahousse) [reply src/entity/Store.ts]: But what's a manager ?

**Comment by hugodelahousse** [comment]
Could you add tests to this ? Like add test fixtures to make sure everything gets created properly, and some simple relationships tests ?

File: src/entity/Store.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrColorado) [reply src/entity/Store.ts]: Someone that can add a promotion with the store name not just a lamda user

[the remark this answers is not in the document]
Comment (hugodelahousse) [reply src/entity/Store.ts]: And where is the differentiation between managers and regular users ? How is it going to be handled in our app ?

File: src/entity/User.ts [hunk 3]
```diff
@@ -15,4 +18,13 @@ export class User {
...
   age: number;
 
+  @OneToMany(type => Vote, vote => vote => vote.user)
+  votes: Vote[];
+
+  @OneToMany(type => Promotion, promotion => promotion.id)
+  promotions: Promotion[];
+
+  @ManyToMany(type => Store, store => store.id)
+  stores: Store[];
```
Comment (hugodelahousse) [inline src/entity/User.ts]: This should be more explicit

File: src/entity/User.ts [hunk 4]
```diff
@@ -15,4 +18,13 @@ export class User {
   @Column()
   age: number;
 
+  @OneToMany(type => Vote, vote => vote => vote.user)
+  votes: Vote[];
+
+  @OneToMany(type => Promotion, promotion => promotion.id)
+  promotions: Promotion[];
```
Comment (hugodelahousse) [inline src/entity/User.ts]: This should be more explicit

File: src/entity/UserList.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { Entity, OneToMany, PrimaryGeneratedColumn, Column, ManyToOne } from 'typeorm';
+import { User } from './User';
+import { ListProduct } from './ListProduct';
+
+@Entity()
+export class UserList {
```
Comment (hugodelahousse) [inline src/entity/UserList.ts]:
```suggestion
export class ShoppingList {
```

File: src/tests/checkVote.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    await voteRepository.save(vote);
+
+    vote = voteRepository.create({
+      promotion,
+      user: user2,
+      upvote: true,
+    });
+    await voteRepository.save(vote);
+
+    const filter: {[key: string]: boolean}[] = [];
```
Comment (hugodelahousse) [inline src/tests/checkVote.ts]:
```suggestion
    const filter = [{upvote: true}];
```

File: src/tests/checkVote.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    filter.push({ upvote: true });
+    const data = await voteRepository.find({
+      where: filter,
+    });
+
+    expect({
+      value: 4,
+    }).to.deep.equal({
```
Comment (hugodelahousse) [inline src/tests/checkVote.ts]: deep is useless here

File: src/tests/checkVote.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    });
+    await voteRepository.save(vote);
+
+    const filter: {[key: string]: boolean}[] = [];
+    filter.push({ upvote: true });
+    const data = await voteRepository.find({
+      where: filter,
+    });
+
+    expect({
```
Comment (hugodelahousse) [inline src/tests/checkVote.ts]:
```suggestion
    expect(data.length).to.be.equal(4);
```

File: src/tests/checkVote.ts [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+      promotion,
+      user: user2,
+      upvote: true,
+    });
+    await voteRepository.save(vote);
+
+    const filter: {[key: string]: boolean}[] = [];
+    filter.push({ upvote: true });
+    const data = await voteRepository.find({
+      where: filter,
```
Comment (hugodelahousse) [inline src/tests/checkVote.ts]:
```suggestion
      where: { upvote: true },
```

**Pull request merged**